src/provisioner/provisioner.py

```diff
@@ -218,22 +195,24 @@ def _get_node_info(cortx_conf: ConfigStore):
     def _provision_components(cortx_conf: ConfigStore, mini_prov_interfaces: list, apply_phase: str):
         """Invoke Mini Provisioners of cluster components."""
         node_id, node_name = CortxProvisioner._get_node_info(cortx_conf)
-        components = cortx_conf.get(f'node>{node_id}>components')
-        if components is None:
-            Log.warn(f"No component specified for {node_name} in CORTX config")
-        num_components = len(components)
         for interface in mini_prov_interfaces:
-            for comp_idx in range(0, num_components):
-                services = cortx_conf.get(
-                    f'node>{node_id}>components[{comp_idx}]>services')
-                service = 'all' if services is None else ','.join(services)
-                component_name = components[comp_idx]['name']
+            component_idx = 0
+            while cortx_conf.get(f'node>{node_id}>components[{component_idx}]>name') is not None:
```

Comment:
Instead of comparing with None, Lets add a new key in the conf store name "node>{node_id}>num_components" so that we can iterate easily.

Reply:
Done.